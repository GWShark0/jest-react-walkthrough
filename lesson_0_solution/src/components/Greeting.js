import React, { Component } from 'react';

class Greeting extends Component {
  render() {
    return (
      <h1 className="greeting">
        Hello, {this.props.name}!
      </h1>
    );
  }
}

export default Greeting;
