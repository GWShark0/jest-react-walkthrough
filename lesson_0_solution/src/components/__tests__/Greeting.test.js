import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Greeting from '../Greeting';

describe('Greeting', () => {

  it('should render', () => {
    const component = renderer.create(<Greeting />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should display name (snapshot)', () => {
    const component = renderer.create(<Greeting name="World" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should display name (enzyme)', () => {
    const wrapper = shallow(<Greeting name="World" />);

    expect(wrapper.text()).toContain('World');
  });

  it('should have a `greeting` className', () => {
    const wrapper = shallow(<Greeting />);

    expect(wrapper.hasClass('greeting')).toBe(true);
  });

  it('should be an <h1>', () => {
    const wrapper = shallow(<Greeting />);

    expect(wrapper.type()).toBe('h1');
  });

});
