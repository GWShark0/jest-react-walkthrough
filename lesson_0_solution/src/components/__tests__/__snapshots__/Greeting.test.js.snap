// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting should display name (snapshot) 1`] = `
<h1
  className="greeting"
>
  Hello, 
  World
  !
</h1>
`;

exports[`Greeting should render 1`] = `
<h1
  className="greeting"
>
  Hello, 
  !
</h1>
`;
